docs/faucets/dd.md

```diff
@@ -0,0 +1,363 @@
+
+We can calculate the maximum mint amount based on the expected L1 block production rate, the expected L1 gas cost of a mint, and the amount of time we expect the handler to be live.
+
+Assume the cost to mint is 5K gas.
+
+Then if a block were completely full of `mint` transactions, the maximum number of mints would be 38M / 5K = 7600.
+
+This means that we could be maximally minting 7600 / 12 = 633 times per second.
+
+Suppose that we expect the handler to be live for a maximum of 5 years.
+
+Then we have a maximum of 5 \* 365 \* 24 \* 60 \* 60 \* 633 ~= 1e11 mints.
+
+Thus, the maximum mint amount is (2^256 - 1) / 1e11 ~= 1e66.
+
+But we'll err on the side of caution and use 1e42.
+
+#### Alternative Design
+
+The primary alternative would be to make a frontend UI that connects to a backend that has the account that is the owner of the token, then maintain a map on the backend to rate-limit different IP addresses, or have users sign up, etc.
+
+The benefit is that if we went with the signup route, we could advertise to those users.
+
+The downside is that that is significantly more infrastructure, and it doesn't necessarily improve the safety/control of the system.
+
+### Staking Asset
+
+The staking asset will be deployed with the same owner as the fee asset. In addition, a `StakingAssetHandler` will be deployed, which will have the minter role on the staking asset.
+
+### Staking Asset Handler
+
+```solidity
+interface IStakingAssetHandler {
+	function addValidator(address _attester, address _proposer) external;
+	function setRollup(address _rollup) external;
+	function setDepositAmount(uint256 _amount) external;
+	function setMinMintInterval(uint256 _interval) external;
+}
+
+contract StakingAssetHandler is IStakingAssetHandler, Ownable {
+	IMintableERC20 public immutable STAKING_ASSET;
+
+	uint256 public depositAmount;
+	uint256 public lastMintTimestamp;
+	uint256 public minMintInterval;
+	IStakingCore public rollup;
+
+	constructor(address _owner, address _stakingAsset, address _rollup, uint256 _depositAmount, uint256 _minMintInterval) Ownable(_owner) {
+		STAKING_ASSET = IMintableERC20(_stakingAsset);
+		depositAmount = _depositAmount;
+		rollup = IStakingCore(_rollup);
+		minMintInterval = _minMintInterval;
+	}
+
+	function addValidator(address _attester, address _proposer) external override onlyOwner {
+		require(block.timestamp - lastMintTimestamp >= minMintInterval, "Min mint interval not met");
+		lastMintTimestamp = block.timestamp;
+		STAKING_ASSET.mint(address(this), depositAmount);
+		rollup.deposit(_attester, _proposer, owner(), depositAmount);
+	}
+
+	function setRollup(address _rollup) external override onlyOwner {
+		rollup = IStakingCore(_rollup);
+	}
+
+	function setDepositAmount(uint256 _amount) external override onlyOwner {
+		depositAmount = _amount;
+	}
+
+	function setMinMintInterval(uint256 _interval) external override onlyOwner {
+		minMintInterval = _interval;
+	}
+
```

Comment:
I would add `setWithdrawer(address _withdrawer)` that can be called by the owner. And not have bot be the owner. Bot should not have admin rights and only minimal access.